sbc_compassion/models/import_letters_history.py

```diff
@@ -348,15 +354,17 @@ def _find_qrcode(self, img, MULTIPAGE, file_,
         if qrcode is None:
             # rotate image
             img = img[::-1, ::-1]
-            # save it for zxing
-            cv2.imwrite(file_png, img)
             # second attempt
             qrcode = zx.decode(file_png, try_harder=True, crop=[
                 int(left), int(top), int(width), int(height)])
             # if the QR code is found
             if qrcode is not None:
+                # save it for zxing
+                cv2.imwrite(file_png, img)
                 # replace the image by the returned one
                 data = self._save_img(MULTIPAGE, file_, img)
+            else:
+                qrcode = zx.decode(file_png, try_harder=True)
```

Comment:
At line 373, add a condition to the search of partners :
`('is_company', '=', False)`